component-runtime-impl/src/main/java/org/talend/sdk/component/runtime/record/SchemaImpl.java

```diff
@@ -514,6 +521,13 @@ public Builder withType(final Type type) {
                 return this;
             }
 
+            @Override
+            public Builder withLogicalType(final SchemaProperty.LogicalType logicalType) {
+                this.logicalType = logicalType;
+                this.props.put(SchemaProperty.LOGICAL_TYPE, logicalType.key());
+                return this;
+            }
```

Comment:
🤔 why did you also put the `logicalType` into the `props` map ?

I ask that because it was not done for `type` or other attributes.

Reply:
Because, the logicalType in the schema is held as a schema property, the logicalType type is only used in the builder.